### ISSUE TITLE
Unable to index a ragged array...

### ISSUE THREAD

**Issue by manor** [timholy/HDF5.jl#242]
If the dataset is a VLEN, the library does not support indexing: 

https://github.com/timholy/HDF5.jl/blob/master/src/plain.jl#L1553

By contrast, transparent access to ragged arrays is supported in h5py: 

http://docs.h5py.org/en/latest/special.html#arbitrary-vlen-data

@simonster pointed out that [read()](https://github.com/timholy/HDF5.jl/blob/master/doc/hdf5.md#reading-and-writing-data) deals with this datatype and I will definitely use it as a workaround, but it is a bit of a bummer to require the loading of the entire dataset in order to access e.g. a single row or even a single element...

**Comment by timholy** [comment]
Agreed this would be a good feature to add. Unfortunately I'm pretty swamped for some time yet, so if you need this soon you might consider whether you are up to trying to implementing it.
